llama_index/embeddings/voyageai.py

```diff
@@ -0,0 +1,59 @@
+"""OpenAI embeddings file."""
+
+from typing import List, Optional
+
+import voyageai
+
+from llama_index.embeddings.base import BaseEmbedding
+
+
+class VoyageEmbedding(BaseEmbedding):
+    """Class for Voyage embeddings.
+
+    Args:
+        model (str): Model for embedding.
+            Defaults to "voyage-01".
+
+        deployment_name (Optional[str]): Optional deployment of model. Defaults to None.
+            If this value is not None, mode and model will be ignored.
+            Only available for using AzureOpenAI.
+    """
+
+    model: str = "voyage-01"
+    voyage_api_key: Optional[str] = None
+
+    def __init__(
+        self,
+        model: str = "voyage-01",
+        voyage_api_key: Optional[str] = None,
```

Comment:
can we please also add `embed_batch_size` and `callback_manager` for users to be able to specify and pass these down in the `super().__init__` call.

Reply:
as an example see [here](https://github.com/run-llama/llama_index/blob/14a059f6318c0b30c7799ac6f882d372aaf6a1d4/llama_index/embeddings/clarifai.py#L68)